parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -253,14 +253,21 @@ retrycmd_get_tarball_from_registry_with_oras() {
 retrycmd_get_access_token_for_oras() {
     retries=$1; wait_sleep=$2; url=$3
     for i in $(seq 1 $retries); do
-        ACCESS_TOKEN_OUTPUT=$(timeout 60 curl -v -s -H "Metadata:true" --noproxy "*" "$url")
-        if [ -n "$ACCESS_TOKEN_OUTPUT" ]; then 
+        response=$(timeout 60 curl -v -s -H "Metadata:true" --noproxy "*" "$url" -w "\n%{http_code}")
```

Comment:
if it's a 4xx error, doesn't IMDS return a json object detailing the error details? I know there's an error_description field you can use to get more detailed error info

I'd probably opt for using `jq` to parse out error_description from the response json and check if it's empty or not, if it's non-empty it should indicate an error

Reply:
on second thought, might be worth following the published retry guidance a little more closely: https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/how-to-use-vm-token#retry-guidance

Reply:
If it is 400 error, it will return json with error details. However, unsure about other 4xx errors. e.g. in acr refresh token part (not IMDS), if it is 401, the boject is empty

About retry time, i extend it for standalone test, will revert it back to guidance value. Sorry for confusion